libclamav_rust/src/alz.rs

```diff
@@ -0,0 +1,551 @@
+            let e: ALZParseError = parse_result.err().unwrap();
+            info!("{}", e.get_description());
+            return false;
+        }
+
+        if let Err(err) = local_fileheader.is_supported() {
+            info!("{err}");
+            return false;
+        }
+
+        if !local_fileheader.is_directory() {
+            /* The is_file flag doesn't appear to always be set, so we'll just assume it's a file if
+             * it's not marked as a directory.*/
+            let res2 = local_fileheader.extract_file(cursor, &mut self.embedded_files);
+            if res2.is_err() {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    #[allow(clippy::unused_self)]
+    fn parse_central_directoryheader(&self, cursor: &mut std::io::Cursor<&Vec<u8>>) -> bool {
+        /*
+         * This is ignored in unalz (UnAlz.cpp ReadCentralDirectoryStructure).
+         *
+         * It actually reads 12 bytes, and I think it happens to work because EOF is hit on the next
+         * read, which it does not consider an error.
+         */
+        let ret = cursor.read_u64::<LittleEndian>();
+        ret.is_ok()
+    }
+
+    #[must_use]
+    pub const fn new() -> Self {
+        Self {
+            embedded_files: Vec::new(),
+        }
+    }
+
+    /// # Errors
+    /// Will return `ALZParseError` if file headers are not correct or are inconsistent.
+    pub fn from_bytes(bytes: &'aa [u8]) -> Result<Self, ALZParseError> {
+        let binding = bytes.to_vec();
+        let mut cursor = Cursor::new(&binding);
+
+        let mut alz: Self = Self::new();
+
+        if !alz.is_alz(&mut cursor) {
+            return Err(ALZParseError::new("No ALZ file header"));
+        }
+
+        //What these bytes are supposed to be in unspecified, but they need to be there.
+        let ret = cursor.read_u32::<LittleEndian>();
+        if ret.is_err() {
+            return Err(ALZParseError::new("Error reading uint32 from file"));
+        }
+
+        loop {
+            let Ok(sig) = cursor.read_u32::<LittleEndian>() else {
+                break;
+            };
+
+            match sig {
+                ALZ_LOCAL_FILE_HEADER => {
+                    if alz.parse_local_fileheader(&mut cursor) {
+                        continue;
+                    }
+                }
+                ALZ_CENTRAL_DIRECTORY_HEADER => {
+                    if alz.parse_central_directoryheader(&mut cursor) {
+                        continue;
+                    }
+                }
+                ALZ_END_OF_CENTRAL_DIRECTORY_HEADER => {
+                    break;
+                    /*This is the end, nothing really to do here.*/
+                }
+                _ => {
+                    #[allow(clippy::uninlined_format_args)]
+                    return Err(ALZParseError::new(format!(
+                        "Parse Error, unrecognized sig = '{:x}'",
+                        sig
+                    )));
```

Comment:
```suggestion
                    return Err(Error::UnrecognizedSig(format!("{:x}", sig)));
```

I love your use of `format!()` with `{:x}`. It was tempting to use the `Parse` enum variant, but that takes a `&str` cause those messages are all constants, so I added an extra enum variant for this one to keep your `format!()`.